---
title: 'Nature Methods on Transcriptome Analysis'
date: 2020-02-04
permalink: /blogs/2020/02/nat-met-on-transcriptome-analysis/
author_profile: false
tags:
  - machine learning
  - transcriptomics
  - computational biology
---

A track of [Nature Methods](https://www.nature.com/nmeth/) papers related to transcriptome analysis and machine learning.


<p align="center">
    <img src="https://hzmaxwell.github.io/images/nature-methods.jpg" width="600"/>
</p>



(Up to Feb 25, 2020)




- Feb 2020
  - Statistical analysis of spatial expression patterns for spatially resolved transcriptomic studies
- Jan 2020
  - Benchmarking algorithms for gene regulatory network inference from single-cell transcriptomic data
- Dec 2019
  - Orchestrating single-cell analysis with Bioconductor
- Nov 2019
  - Probabilistic cell typing enables fine mapping of closely related cell types in situ
  - Fast, sensitive and accurate integration of single-cell data with Harmony
- Oct 2019
- Exploring single-cell data with deep multitasking neural networks
- Sept 2019
  - Supervised classification enables rapid annotation of cell atlases
  - Probabilistic cell-type assignment of single-cell RNA-seq for tumor microenvironment profiling
- Aug 2019
  - Data denoising with transfer learning in single-cell transcriptomics
- Jul 2019
  - scGen predicts single-cell perturbation responses
  - Joint analysis of heterogeneous single-cell RNA-seq dataset collections
- Jun 2019
  - Pathway-level information extractor (PLIER) for gene expression data
- May 2019
  - Benchmarking single cell RNA-sequencing analysis pipelines using mixture control experiments
- Apr 2019
  - cisTopic: cis-regulatory topic modeling on single-cell ATAC-seq data
- Mar 2019
  - Scalable analysis of cell-type composition from single-cell transcriptomics using deep recurrent learning
  - Selene: a PyTorch-based deep learning library for sequence data
  - Deep-learning augmented RNA-seq analysis of transcript splicing







